crates/sats/src/buffer.rs

```diff
@@ -20,6 +20,8 @@ pub enum DecodeError {
     InvalidTag { tag: u8, sum_name: Option<String> },
     /// Expected data to be UTF-8 but it wasn't.
     InvalidUtf8,
+    /// Expected the byte to be 0 or 1 to be a valid bool.
+    InvalidBool(u8),
```

Comment:
I wonder if there's some better way we can encode the set of errors "value is not valid at expected type." I don't see much of a difference between `InvalidTag`, `InvalidUtf8` and `InvalidBool`.

This is not a request for any changes, just some musing.